### PR TITLE
fix .toString() state (#531)

### PR DESCRIPTION
**Pull request by louiscenter**
Fixed a bug where `.toString()` method was not extending the `state` object properly: https://github.com/choojs/choo/issues/531

### CODE REVIEW

**Comment by louiscenter** [comment]
I have updated the fix from:

```js
this.state = xtend(state, this.state || {})
```

to

```js
this.state = xtend(this.state || {}, state || {})`
```

I found that if `state` was passed in as the first argument of `xtend()`, `this.state` wouldn't be updated properly if there were successive calls to `app.toString()` in the same node script.

My understanding is that the properties of `xtend()`'s second argument will take precedent over the first.

**Comment by yoshuawuyts** [comment]
Oh yeah, now I remember - the approach here might leak values from previous
runs which is not good :/ - should find a way to create a new state on each
run - perhaps reset the state to the original state after the string has
been created or smth?

**Comment by louiscenter** [comment]
would something like this work?

```js
Choo.prototype.toString = function (location, state) {
  this.state = xtend(this.state || {}, state || {})

  assert.notEqual(typeof window, 'object', 'choo.mount: window was found. .toString() must be called in Node, use .start() or .mount() if running in the browser')
  assert.equal(typeof location, 'string', 'choo.toString: location should be type string')
  assert.equal(typeof this.state, 'object', 'choo.toString: state should be type object')

  this.state.query = nanoquery(location)
  var html = this.router(location)
  assert.ok(html, 'choo.toString: no valid value returned for the route ' + location)

  var str = html.toString()
  this.state = { events: this._events }

  return str
}
```

File: index.js [hunk 1]
```diff
@@ -176,7 +176,7 @@ Choo.prototype.mount = function mount (selector) {
 }
 
 Choo.prototype.toString = function (location, state) {
-  this.state = xtend({ events: xtend(this._events) }, this.state || {})
+  this.state = xtend(this.state || {}, state || {})
```
Comment (yoshuawuyts) [inline index.js]: `this.state` always exists; think we should be able to get away with `xtend(this.state, state || {})`

**Comment by yoshuawuyts** [comment]
`v6.0.0-4`

**Pull request merged**